extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/security/HelloResource.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.resteasy.reactive.server.test.security;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+import org.jboss.resteasy.reactive.server.ServerExceptionMapper;
+
+import io.quarkus.security.UnauthorizedException;
+
+@Path("/hello")
+public class HelloResource {
+    @GET
+    public String hello() {
+        return "hello";
+    }
+
+    @ServerExceptionMapper
+    public Response unauthorizedExceptionMapper(UnauthorizedException unauthorizedException) {
```

Comment:
What I don't understand, why is `unauthorizedExceptionMapper` requiring a security check while your test verifies that `GET /hello` returns 200 ?

Reply:
Never mind, you'd like to make sure it is invoked

Reply:
That said, it seems strange the security related modifications are required because a mapper is declared in the resource class, it is not a JAX-RS/endpoint method...

Reply:
I'm not sure if I got your point, but I'll try to explain this: here is an example with `@ServerExceptionMapper` that is declared as resource method. Currently all public non static non constructor resource methods are secured. I used the mapper method as it verifies the issue, however I could as well declare plain public method and call it within endpoint -> same issue! Let see tests that I added for RESTEasy Classic, they show that.

Reply:
And no, I added it there as I want to make sure it's invoked and returns value correctly. Without the fix, method don't return anything as an exception is thrown.